app/src/main/java/com/dsvag/currencyexchanger/data/adapters/CoinAdapter.kt

```diff
@@ -16,18 +20,27 @@ class CoinAdapter() : RecyclerView.Adapter<CoinAdapter.CoinViewHolder>() {
         val inflater = LayoutInflater.from(parent.context)
 
         return CoinViewHolder(
-            RowCoinBinding.inflate(inflater, parent, false)
+            RowCoinBinding.inflate(inflater, parent, false),
+            moveToTop = {
+                data.add(0, data.removeAt(it))
+                notifyItemMoved(it, 0)
+                recyclerView.scrollToPosition(0)
+            },
+            {
```

Comment:
А че этот неименованный?
```suggestion
            reprice = {
```
А вообще я бы вынес это и `moveToTop` в метод, а сюда передавал бы ссылки на метод вот так: `::reprice`